### ISSUE TITLE
Cross staff notation small issues

### ISSUE THREAD

**Issue by Omicronrg9** [Jojo-Schmitz/MuseScore#322]
Hi again!

I have noticed that note-heads overlap one another when using cross-staff notation _(see 1st IMG below)_. The left hand has the same pattern in both halves of the bar, but using cross-staff notation makes the 8th note head overlap the half note one & additionally, some stacatto symbols get misaligned/misplaced _(second image & third image show the exact same bar but with cross-staff notation and without)_.
![image](https://github.com/Jojo-Schmitz/MuseScore/assets/13071402/affa8860-58c5-42f9-bf48-846c7564434c)

![image](https://github.com/Jojo-Schmitz/MuseScore/assets/13071402/548b9537-b318-4051-aaee-4074f6e89b4e)
![image](https://github.com/Jojo-Schmitz/MuseScore/assets/13071402/da518f43-88a6-4a78-ad4a-127ff8bd6535)

I updated MS3.7 recently _(MuseScore-3.7.0.7870110174-x86_64)_

**Comment by Jojo-Schmitz** [comment]
Does this issue exist in a) 3.6.2 (i.e. is is a regression in 3.7) and b) 4.2.1 too?

**Comment by Omicronrg9** [comment]
MS4.2.1: No 
![image](https://github.com/Jojo-Schmitz/MuseScore/assets/13071402/b461c190-2881-488b-89a5-adb229fe3942)

MS 3.6.2: Yes. 
![image](https://github.com/Jojo-Schmitz/MuseScore/assets/13071402/7d9d99a1-ca27-4467-b3d2-ea547c70a16d)

**Comment by Jojo-Schmitz** [comment]
OK, so now we'd need to find which change in Mu4 fixed that, in order to be able to port it back to Mu3

**Comment by Jojo-Schmitz** [comment]
Any of https://github.com/musescore/MuseScore/pulls?q=is%3Apr+sort%3Aupdated-desc+is%3Aclosed+in%3Atitle+cross+staff  might be candidates

**Comment by Jojo-Schmitz** [comment]
When in doubt, try it out...

**Comment by Jojo-Schmitz** [comment]
I don't think that fits y is though, as that was fixing a 3.7 regression whereas they s here happens in 3.6 too
